src/v/coproc/service.cc

```diff
@@ -157,13 +167,29 @@ ss::future<disable_response_code> service::remove(script_id id) {
       "Incoming request to disable coprocessor with script_id {}",
       id);
     using drc = disable_response_code;
-    return _router
-      .map_reduce0(
-        [id](router& r) { return r.remove_source(id); },
-        false,
-        std::logical_or<>())
-      .then([](bool r) {
-          return r ? drc::success : drc::script_id_does_not_exist;
+    return _router.map_reduce0(
+      [id](router& r) {
+          try {
+              return r.remove_source(id).then([](bool removed) {
+                  return removed ? drc::success : drc::script_id_does_not_exist;
+              });
+          } catch (const std::exception& ex) {
+              vlog(
+                coproclog.warn,
```

Comment:
that's an error no? how is this recoverable ? 

i think this should be fatal IMO. 

how can you recover from this error ?

Reply:
if you look below, you'll see this returns internal_error, which will eventually be boiled down and returned to the client. It should interpret this as a failed registration to the topic.